src/OpenApi/src/Services/OpenApiConstants.cs

```diff
@@ -14,19 +14,19 @@ internal static class OpenApiConstants
     internal const string SchemaId = "x-schema-id";
     internal const string RefId = "x-ref-id";
     internal const string DefaultOpenApiResponseKey = "default";
-    // Since there's a finite set of operation types that can be included in a given
-    // OpenApiPaths, we can pre-allocate an array of these types and use a direct
+    // Since there's a finite set of HTTP methods that can be included in a given
+    // OpenApiPaths, we can pre-allocate an array of these methods and use a direct
     // lookup on the OpenApiPaths dictionary to avoid allocating an enumerator
     // over the KeyValuePairs in OpenApiPaths.
-    internal static readonly OperationType[] OperationTypes = [
-        OperationType.Get,
-        OperationType.Post,
-        OperationType.Put,
-        OperationType.Delete,
-        OperationType.Options,
-        OperationType.Head,
-        OperationType.Patch,
-        OperationType.Trace
+    internal static readonly HttpMethod[] HttpMethods = [
+        HttpMethod.Get,
+        HttpMethod.Post,
+        HttpMethod.Put,
+        HttpMethod.Delete,
+        HttpMethod.Options,
+        HttpMethod.Head,
+        HttpMethod.Patch,
+        HttpMethod.Trace
```

Comment:
I've seen an issue that there's a new QUERY method coming soon, so that might need adding shortly.

Reply:
https://github.com/dotnet/runtime/issues/114489#issuecomment-2807076031

Reply:
AFAIK, the OpenAPI spec doesn't yet support all HTTP methods. There's been discussion of loosening this requirement in the past but at the moment it only supports this fixed set.